droidlet/memory/memory_filters.py

```diff
@@ -471,10 +471,12 @@ def handle_output(self, agent_memory, query, memids):
                         )
                     )
                 for m in memids:
-                    values_dict[m].append(get_property_value(agent_memory, m, aname))
-            if len(attribute_name_list) == 1:
-                for m in values_dict:
-                    values_dict[m] = values_dict[m][0]
+                    prop_vals = get_property_value(agent_memory, m, aname, get_all=True)
```

Comment:
this should be propagated from .search().    To be backwards compatible, make the default False for now; and in the nsm code use .search(query, get_all=True)